src/main/java/gui/DashboardWindow.java

```diff
@@ -181,18 +180,17 @@ private void jButtonLoadScenarioAction() {
                 warningWindow.setVisible(true);
             } else {
 
+                   factory = OntologyLogic.getFactory(pathToOwlFile);
```

Comment:
Ogólnie naprawa polega na tym, że zawsze robię nowe factory jak wczytuje z pliku. I w funkcji  OntologyLogic.getModelFromOntology(factory, jTextFieldScenarioName.getText()); jest używane tylko to jedno factory już nie ma żadnego innego tworzenia factory po drodze. Problem jest taki, że jest to jednak czasochłonne, ale się nie da tego uniknąć.

Reply:
Ogólnie to problem z pewnym rodzajem cachowaniem factory polegałby na tym, że faktycznie można nie odświeżać factory, jeśli wcześniej operacje do ontologii przeprowadzaliśmy tylko przez to jedno factory (co może być spełnione) to w środku scenarioReadera dla każdego pobrania obiektu z ontologii musiałoby być takie zabezpieczenie
![image](https://user-images.githubusercontent.com/43828836/100763223-2b379880-33f5-11eb-8745-bfa59ca9222e.png)

Reply:
Co jest dosyć kiepskie, ale inaczej polecą nulle